motto/agents/retrieval_agent.py

```diff
@@ -0,0 +1,115 @@
+import shutil
+
+import chromadb
+import numpy as np
+import os
+import pathlib
+import concurrent.futures
+
+from .agent import Agent
+from .agent import Response
+from .data_processors import OpenAIEmbeddings, DocProcessor
+from hyperon import *
+
+
+def fix_string(value):
+    value = value.strip()
+    if len(value) > 1 and value[0] == '"' and value[-1] == '"':
+        return value[1:-1]
+    return value
+
+
+class RetrievalAgent(Agent):
+    max_length = 2270
+
+    def __init__(self, data_source, chunk_token_size, docs_count, data_dir):
+
+        if isinstance(data_source, GroundedAtom):
+            data_source = fix_string(repr(data_source))
+        if not (os.path.isfile(data_source) or os.path.isdir(data_source)):
+            raise AttributeError("data_source should be file or folder")
+
+        if not os.path.exists(data_source):
+            return
+        if isinstance(chunk_token_size, GroundedAtom):
```

Comment:
With `unwrap=True`, grounded atoms should be unwrapped into their values before getting here. Is there a case, when they pass here?